### PR TITLE
Update parser signature for `aiida-shell` v0.8, extend ShellJob docs, use `prepare_shell_job_inputs`

### PR DESCRIPTION
**Pull request by GeigerJ2**
Still needs some work, but already creating the PR now to see if CI runs through.

### CODE REVIEW

File: aiida_workgraph/engine/utils.py [hunk 1]
```diff
@@ -152,11 +153,19 @@ def prepare_for_shell_task(task: dict, kwargs: dict) -> dict:
...
             nodes[key[6:]] = kwargs.pop(key)
     metadata.update({"call_link_label": task["name"]})
+
+    arguments = kwargs.pop("arguments", [])
+    # Split on whitespace if arguments given as `str`
+    if isinstance(arguments, str):
+        arguments = shlex.split(arguments)
+    else:
+        lang.type_check(arguments, list, allow_none=True)
+
```
Comment (superstar54) [inline aiida_workgraph/engine/utils.py]: Hi @GeigerJ2 , thanks for the work, the idea is to call the new `prepare_shell_job_inputs` (from aida-shell) inside this `prepare_for_shell_task` function, to replace lines 142-150 (prepare code etc), 157-162 (slit argument). Thus we don't need to manually update this function if aiida-shell updates.

Comment (GeigerJ2) [reply aiida_workgraph/engine/utils.py]: Ah, OK, I see. Yeah, I was also wondering why we duplicate some code from that function here, but thought there might be a reason to handle some things differently in WorkGraph.

Comment (GeigerJ2) [reply aiida_workgraph/engine/utils.py]: Wasn't aware of the relevant [PR](sphuber/aiida-shell#103) in `aiida-shell`. Now using the `prepare_shell_job_inputs` helper function, and just updating the `kwargs` `nodes` and `metadata` as required.

**Comment by GeigerJ2** [comment]
Finally, CI passing 😅 OK, this should be ready for review, @superstar54 :)

**Comment by superstar54** [comment]
Hi @GeigerJ2 , thanks for the work! I modified the `prepare_for_shell_task` function because some of the code is already covered by the `prepare_shell_job_inputs`.

**Pull request merged**